internal/clusterfeature/clusterfeatureadapter/workflow/cluster_feature_deactivate_activity.go

```diff
@@ -28,17 +28,17 @@ type ClusterFeatureDeactivateActivityInput struct {
 }
 
 type ClusterFeatureDeactivateActivity struct {
-	features clusterfeature.FeatureRegistry
+	features clusterfeature.FeatureOperatorRegistry
 }
 
-func MakeClusterFeatureDeactivateActivity(features clusterfeature.FeatureRegistry) ClusterFeatureDeactivateActivity {
+func MakeClusterFeatureDeactivateActivity(features clusterfeature.FeatureOperatorRegistry) ClusterFeatureDeactivateActivity {
```

Comment:
```suggestion
func MakeClusterFeatureDeactivateActivity(features clusterfeature.FeatureOperatorRegistry) *ClusterFeatureDeactivateActivity {
```

Reply:
I think that this is unnecessary. As a performance optimization it would be premature. As for instance sharing, you could make that decision on the caller's side by taking the address of the returned instance and passing that around. Also, instance sharing is only relevant if you mutate the instance, which this implementation does not do. See also: https://stackoverflow.com/a/23551970/578530